docs/proposals/20200827-history.md

```diff
@@ -0,0 +1,175 @@
+
+```bash
+kconnect ls --provider eks -o table
+
+| ID                        | Providers | Identity | Alias       | Config     | Flags                      |
+|===========================|===========|==========|=============|============|============================|
+|MTIzNDU2Nzg5L215Y2x1c3RlcjE| EKS       | SAML     | dev-eks     | (default)  | username=bob@somewhere.com |
+|                           |           |          |             |            | idp-provider=GoogleApps    |
+|---------------------------|-----------|----------|-------------|------------|----------------------------|
+
+```
+
+
+#### Using the history
+
+Using a history item will require changes to the `use` command. Its proposed that the `use` command be changed to support the following ux
+
+```bash
+# One of the following 2 commands would be used
+kconnect use history --id MTIzNDU2Nzg5L215Y2x1c3RlcjE
+kconnect use history --alias dev-eks
+
+Using history item MTIzNDU2Nzg5L215Y2x1c3RlcjE (eks/saml)
+
+Enter your password: *******************
+
+saving your settings to $HOME/.kconnect/config.yaml
+
+you can run kubectl commands now!
+
+```
+
+The above UX implies that a new provider be created called `history`. This may not fit the current code and will need to be investigated at implementation time. If its not possible then a new command will be introduced:
+
+```bash
+kconnect use-history --id MTIzNDU2Nzg5L215Y2x1c3RlcjE
+kconnect use-history --alias dev-eks
+```
+
+### Changes to `kconnect use`
+
+To support the alias functionality we will need to make some changes to the `use` command to allow the user to optionally enter an alias.Its proposed that an additional flag be included `-a/--alias` that the user can specify when running the command. If the flag isn't specified and we are running in interactive mode then the user will be asked to enter an alias.
+
+If an alias is specified via flag or user entry then the history will be checked to see if the alias has already been used. If it has already been used then an error will be returned to the user.
+
+### Changes to providers
+
+The existing providers will need to be changed so that they return a unique identifier for a cluster than is accessible via that provider. As an example a EKS cluster in AWS can be uniquely identied via its ARN or via a account number/cluster name combination. It would be up to the provider to return this provider unique identifier in the [Cluster.ID](https://github.com/fidelity/kconnect/blob/main/pkg/provider/types.go#L85) field.
+
+### Additional Details
+
+We could also consider adding a new command called `alias` that has an number of subcommands to allow the user to maintain the aliases. For example:
+
+```bash
+# Add an alias to a history item
+kconnect alias add --id MTIzNDU2Nzg5L215Y2x1c3RlcjE --alias dev-eks
+
+# Delete an alias from a history item
+kconnect alias rm --id MTIzNDU2Nzg5L215Y2x1c3RlcjE
+
+# List all the aliases an their history ids
+kconnect alias ls
```

Comment:
A clear command would be useful as well:
kconnect alias clear
or
kconnect aliam rm --all